pages/add/form/validator.js

```diff
@@ -53,6 +58,39 @@ const Validator = {
         }
       }
     };
+  },
+  imageSizeValidator() {
+    return {
+      error: `File size is over 2MB.`,
+      validate: function(value) {
+        if (!value) {
+          return;
+        }
+
+        // there's no file size limit on the backend
+        // but it's still good that we enforce a size limit (2MB) on client side
+        let base64String = value.base64;
+        let sizeLimit = Math.pow(2, 20) * 2; // 2MB in bytes
```

Comment:
I'd kind of say "let's just use `2097152` because this is literally a fixed number, no need for computation"